Adding Mailjet service

joefearnley opened the pull request:
Adding Mailjet service. Once (if) merged, I will update the README on how to utilize Mailjet.

joefearnley committed:
Adding Mailjet service

peter279k commented:
I agree with this merge, but some files are conflicted.
It seems that some files does not update the new commit.
Would you use ```git pull``` to solve the conflicts?

Thanks.

joefearnley committed:
Syncing changes from original repo

joefearnley commented:
Hi @peter279k, I have updated my forked repo and added back the changes. I will send another pull request. Unfortunately, I do not have the access to pull from the main repo so I had to do it manually. Let me know if I missed anything.

peter279k commented:
Hi @joefearnley. Did you really try the ```git pull https://github.com/peter279k/serviceBundle``` command?
I think the pull command is will be fine.

Thanks.

joefearnley commented:
Hi @peter279k, 

I don't think I can pull from your repo and write back without write access. I forked the repo, copied everything over,  and submitted a pull request. If there is anything missing, let me know and I will add it to the repo.

peter279k commented:
Hi @joefearnley, you can. I fork many repos and pull commit from the orginal repos to synchronize the commits for the many times.
You just try the pull command from original repo and push the latest commit to your forked repo.
And this PR will be updated.

Thanks.

peter279k commented:
Here is the steps you can refer:
```
git pull https://github.com/peter279k/serviceBundle
# commit or fix the conflicts
git commit -a
git push https://github.com/joefearnley/serviceBundle
```
This will be worked successfully!

joefearnley committed:
attempting merge conflicts

peter279k commented:
Hi @joefearnley, can you remove the ```composer.lock```?
The ```composer.lock``` cached the dependencies cause Travis CI build error.

Thanks.

joefearnley committed:
Removing composer.lock

peter279k commented on tests/serviceBundle/MailjetTest.php:
```diff
@@ -0,0 +1,91 @@
+class MailjetTest extends TestCase
+{
+    /** @test */
+    public function isTypeOfMailgun()
```
Please remove the annotation and use the method named ```testIsTypeOfMailjet```.

peter279k commented on tests/serviceBundle/MailjetTest.php:
```diff
@@ -0,0 +1,91 @@
+    /** @test */
+    public function isTypeOfMailgun()
+    {
+        $mailjetService = (new ServiceFactory)->create('mailjet');
```
The ```mailjet``` seems to be wrong.
It should be ```Mailjet```.

peter279k commented on tests/serviceBundle/MailjetTest.php:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    /** @test */
+    public function canSendEmail()
```
This should remove annotation and use the ```testCanSendEmail``` method name.

peter279k commented on tests/serviceBundle/MailjetTest.php:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    /** @test */
+    public function getsErrorWhenCredentialsAreInvalid()
```
It should remove the annotation and use the ```testGetsErrorWhenCredentialsAreInvalid``` to be the method name.

peter279k commented:
I write the request changes for this PR.
I notice that the unit testing error during the Travis CI build.
Perhaps you can use the command ```composer test``` after submitting the new commit.

Thanks.

joefearnley committed:
Fixing PR issues

joefearnley commented:
Hi @peter279k, I think I have all of the code updated per your request. If not, please let me know.

Thanks!

peter279k commented:
Hi @joefearnley, thank you for your PR.
And you learn how to synchronize the forked repo from the original repo. 

Thank you for your contribution again!

joefearnley commented:
@peter279k Any time!
